decompiler/structures/ast/ast_nodes.py

```diff
@@ -187,6 +187,10 @@ def replace_variable(self, replacee: Variable, replacement: Variable) -> None:
         """Replaces each occurrence of the given variable replacee by the variable replacement in the given AST-node."""
         pass
 
+    def replace_variables(self, replacement_dictionary: dict[Variable, Variable]) -> None:
+        for replacee, replacement in replacement_dictionary.items():
+            self.replace_variable(replacee, replacement)
+
```

Comment:
Two short question: 

1. If you have a binary-operation, say `a + b` that occurs in two different instructions. Then renaming it in one of these instructions, still renames it in the other instruction, right? By handing the dictionary for the renaming, I guess this should work and not lead to a problem. Is this behavior wanted?

2. The renaming-dictionary can be quite large and if we have long instructions, we "iterate" over them multiple times and due to the recursive structure this seems to be quite inefficient. How bad is the runtime overhead? Could it make sense to first get all occurring variables and only try to replace them? Or is the runtime for this negligible?

Reply:
Sry my bad. 
That's part of the old code which is not used anymore. 
Will be removed as dead code.